frozen-abi/macro/src/lib.rs

```diff
@@ -395,19 +409,26 @@ fn frozen_abi_enum_type(input: ItemEnum, expected_digest: &str) -> TokenStream {
 #[cfg(RUSTC_WITH_SPECIALIZATION)]
 #[proc_macro_attribute]
 pub fn frozen_abi(attrs: TokenStream, item: TokenStream) -> TokenStream {
-    let args = parse_macro_input!(attrs as AttributeArgs);
     let mut expected_digest: Option<String> = None;
-    for arg in args {
-        match arg {
-            NestedMeta::Meta(Meta::NameValue(nv)) if nv.path.is_ident("digest") => {
-                if let Lit::Str(lit) = nv.lit {
-                    expected_digest = Some(lit.value());
-                }
-            }
-            _ => {}
+    let attrs_parser = syn::meta::parser(|meta| {
+        if meta.path.is_ident("digest") {
+            expected_digest = Some(meta.value()?.parse::<LitStr>()?.value());
         }
-    }
-    let expected_digest = expected_digest.expect("the required \"digest\" = ... is missing.");
+        Ok(())
+    });
+    parse_macro_input!(attrs with attrs_parser);
+
+    let expected_digest = match expected_digest {
+        Some(expected_digest) => expected_digest,
+        None => {
+            return Error::new_spanned(
```

Comment:
:heart: over my lazy `panic!()`. :)

Reply:
I figured, it should provide better errors, as it includes the location information :)
And the right API was already used 5 lines below.
It only took me some time to figure out how a `TokenStream` can be converted into `quote::ToTokens` that `syn::Error::new_spanned` needs.
The `proc_macro`/`proc_macro2`/`quote`/`syn` dance can be a bit non-obvious sometimes.

Reply:
If I remove the `digest` property, the error I see is actually this:

![image](https://user-images.githubusercontent.com/694419/233750308-fdfe8e6c-8c0f-4e18-a23f-0b55b2aea84d.png)

Reply:
> The `proc_macro`/`proc_macro2`/`quote`/`syn` dance can be a bit non-obvious sometimes.

so true. i remembered the struggle at the time of writing these code... xD